### DISCUSSION TITLE
how to limit output when line length is long

### DISCUSSION THREAD

**Discussion by jsharpe-ctm** [BurntSushi/ripgrep#2035] [Q&A] [answered]
if ripgrep finds something on a line that is 8 billion characters long (minified javascript, for example), it monopolizes the output as all 8 billion of those characters are wrapped into 7 million lines of output and makes it difficult to scroll past all of that.

Is it possible to limit output in cases like this?  If something matches on the 7 thousandth character I'm probably not interested in it.

Thanks!

**Comment by BurntSushi** [comment, marked as answer]
Yes. The `-M/--max-columns` flag will limit the length of each line printed.

**Reply by jsharpe-ctm** [reply]
ah, awesome! ty!

**Reply by stdedos** [reply]
Is it possible to smartly auto-detect the width of the terminal, and stop at that point? (so that each match is one line, "no matter what")

**Reply by atomicmattie** [reply]
> Is it possible to smartly auto-detect the width of the terminal, and stop at that point? (so that each match is one line, "no matter what")

So that's a _little_ tricky. The naïve way would be 

```
rg word -M "${COLUMNS}"
```

But the problem is that, by default, line numbers are printed immediately before the text (and are super useful!), so they'll still overflow a line. They also have varying widths, so it's not just a simple math problem. You could try hoping they're no more than, say, five characters:

```
rg word -M "$((COLUMNS-5))"
```
